web/message_bar.css

```diff
@@ -0,0 +1,174 @@
+	--message-bar-close-button-border-radius: 4px;
+	--message-bar-close-button-border: none;
+	--message-bar-close-button-hover-bg-color: rgb(21 20 26 / 0.14);
+	--message-bar-close-button-active-bg-color: rgb(21 20 26 / 0.21);
+	--message-bar-close-button-focus-bg-color: rgb(21 20 26 / 0.07);
+
+	@media (prefers-color-scheme: dark) {
+		--message-bar-close-button-hover-bg-color: rgb(251 251 254 / 0.14);
+		--message-bar-close-button-active-bg-color: rgb(251 251 254 / 0.21);
+		--message-bar-close-button-focus-bg-color: rgb(251 251 254 / 0.07);
+	}
+
+	@media screen and (forced-colors: active) {
+		--message-bar-close-button-color: ButtonText;
+		--message-bar-close-button-border: 1px solid ButtonText;
+		--message-bar-close-button-hover-bg-color: ButtonText;
+		--message-bar-close-button-active-bg-color: ButtonText;
+		--message-bar-close-button-focus-bg-color: ButtonText;
+		--message-bar-close-button-color-hover: HighlightText;
+	}
+
+	display: flex;
+	position: relative;
+	padding: 12px 8px 12px 0;
+	flex-direction: column;
+	justify-content: center;
+	align-items: center;
+	gap: 8px;
+	align-self: stretch;
+
+	border-radius: 4px;
+
+	border: 1px solid var(--message-bar-border-color);
+	background: var(--message-bar-bg-color);
+	color: var(--message-bar-fg-color);
+
+	> div {
+		display: flex;
+		padding-inline-start: 16px;
+		align-items: flex-start;
+		gap: 8px;
+		align-self: stretch;
+		padding-inline-end: 32px;
+
+		&::before {
+			content: "";
+			display: inline-block;
+			width: 16px;
+			height: 16px;
+			mask-image: var(--message-bar-icon);
+			mask-size: cover;
+			background-color: var(--message-bar-icon-color);
+		}
+
+		> div {
+			display: flex;
+			flex-direction: column;
+			align-items: flex-start;
+			gap: 8px;
+			flex: 1 0 0;
+
+			.title {
+				font-size: 13px;
+				font-weight: 590;
+			}
+
+			.description {
+				font-size: 13px;
+			}
+		}
+	}
+
+	.closeButton {
+		position: absolute;
+		width: 32px;
+		height: 32px;
+		inset-inline-end: 8px;
+		inset-block-start: 8px;
+		background: none;
+		border-radius: var(--message-bar-close-button-border-radius);
+		border: var(--message-bar-close-button-border);
+
+		&::before {
+			content: "";
+			display: inline-block;
+			width: 16px;
+			height: 16px;
+			mask-image: var(--closing-button-icon);
+			mask-size: cover;
+			background-color: var(--message-bar-close-button-color);
+		}
+
+		&:is(:hover, :active, :focus)::before {
+			background-color: var(--message-bar-close-button-color-hover);
+		}
+
+		&:hover {
+			background-color: var(--message-bar-close-button-hover-bg-color);
+		}
+
+		&:active {
+			background-color: var(--message-bar-close-button-active-bg-color);
+		}
+
+		&:focus {
+			background-color: var(--message-bar-close-button-focus-bg-color);
+		}
+
+		> span {
+			display: inline-block;
+			width: 0;
+			height: 0;
+			overflow: hidden;
+		}
+	}
+}
+
+#editorUndoBar {
+	--message-bar-icon: url(images/secondaryToolbarButton-documentProperties.svg);
```

Comment:
That SVG file seems like a strange choice here, too much copy-paste?

Reply:
It was just the closest to the spec that I could find in the existing codebase :sweat_smile: 

I should replace it with a new appropriate SVG but I'd been putting it off until now. I considered downloading the SVG directly from Figma and using that but it has some issues too. I'll ask the UX folks if they have any suggestions here.

Reply:
Verified with the folks from the design team that it's a close enough icon to use and the reusability makes it an even nicer choice.